UI Dev

zacharyhalvorson opened the pull request:
## Resources
- [Link to design spec](http://miscreant-belief.surge.sh)
- [Link to design files and assets](https://drive.google.com/drive/folders/0By6XUcuNmBiAMmExVFNva2ZUc0E?usp=sharing)
- [Link to progressive-web-sdk PR](https://github.com/mobify/progressive-web-sdk/pull/226)
## Instructions
- `npm install`
- `npm link`
- `cd` into `progressive-web-sdk#sdk-docs`
- `npm link react-styleguidist`
- `npm run docs:dev`
## MVP List
- [ ] @lokimeyburg images from confluences need to be loading
- [ ] @lokimeyburg Design docs from confluence are cleaned of all weirdness from export
- [x] App bar positioned properly on detail page
- [x] background colour changes on detail page
- [x] 2-column layout
- [x] table styles
- [x] typeface
- [x] sidebar dismisses on fullscreen detail pages (https://github.com/mobify/progressive-web-sdk/pull/226/commits/fb5e99effbd9b41a88829a2312b087dfca42a9a8) 
- [x] Migrate description from README.md to jsdoc comment in component files
- [x] type styles are correct
- [x] home page directs to components
- [x] Components with no jsx file show up in the list
  - [x] @fractaltheory single column layout
- [x] @jeffkamo close icon
- [ ] sidebar style aligns to mocks
- [x] @fractaltheory Component CSS is imported properly
- [x] @jeffkamo Add easier to identify/find classnames
## Responsive Support
- [ ] collapsing 2-column layout
- [ ] sidebar flyout navigation
## Stretch Goals
- [ ] Sketch Slices in

Commit:
Port changes from SDK repo

Commit:
Fix module resolve error with hot reloading

Commit:
Readme update

Commit:
Allow for npm link development

I think this should work…

Commit:
Nuke a bunch of styles

Commit:
Changes

Commit:
Nuke tags

zacharyhalvorson committed:
Pin webpack dep to 1.x

avoid webpack errors

Commit:
Fix lookup for design readme .md files

zacharyhalvorson committed:
search bar styling

zacharyhalvorson committed:
Added in the components header

zacharyhalvorson committed:
Fixed the appear to the top

zacharyhalvorson committed:
Removed the styleguidist footer

zacharyhalvorson committed:
removed other footer

zacharyhalvorson committed:
Removed extra main tag

zacharyhalvorson committed:
little padding on bottom of list

zacharyhalvorson committed:
add in the detail page app bar

zacharyhalvorson committed:
new close icon

Commit:
Fix up app bar DOM position in detail mode

Commit:
Fix background color in detail mode

Commit:
Hook up keyboard shorcut for search bar

zacharyhalvorson committed:
multi-column layout on desktop

Commit:
Fix detail-page top padding

Commit:
Semicolons

Commit:
Merge remote-tracking branch 'sapegin/master' into ui-dev

# Conflicts:
#	package.json
#	scripts/config.js
#	src/rsg-components/Layout/Layout.js
#	src/rsg-components/Props/Props.js
#	src/rsg-components/TableOfContents/TableOfContents.js

Commit:
Update Readme.md

zacharyhalvorson committed:
Remove styling from styleguidist for headings

Commit:
Remove slightly confusing console.log

- when run in an SDK docs instance

Commit:
Move code description above propList

Commit:
Fix f to search in Safari

- doesn’t have key property!

Commit:
Import components without JSX

Commit:
Refactor design markdown import

Commit:
Single column layout

Commit:
Scroll to top on component render

jeffkamo committed:
Add "Close" icon, SVG and loader to styleguidist

jeffkamo commented:
@fractaltheory @zacharyhalvorson Added the close icon, but after pulling the changes, you will need to `npm install` to get the SVG loader.

Commit:
Insert classes for markdown code css

jeffkamo commented on src/rsg-components/Components/Components.js:
```diff
@@ -27,23 +55,67 @@ export default class Components extends Component {
+		const isListPage = this.props.sidebar;
+		const componentSectionClasses = classNames('w-100 mw8 center ph3', {
+			'pt-appbar-searchbar': isListPage,
+			'pt-appbar': !isListPage
```
@zacharyhalvorson what does this `pt-` prefix mean?

jeffkamo replied:
oh it's a custom tachyon class for `padding-top`... gotcha.

jeffkamo committed:
clean a bit of component template's namings

jeffkamo committed:
Add identifier classnames

Much like Mobify’s standard process. `rgs-` refers to “react style
guide” which should clarify where that component (and therefor any
tachyon classes) is coming from.

jeffkamo committed:
Fix render header functions and rename to make sense

zacharyhalvorson committed:
Modify component list header classes

zacharyhalvorson committed:
reconfigure the spacing on the list page

zacharyhalvorson committed:
tweaks to search bar styling

zacharyhalvorson committed:
Add escape keypress shortcut

zacharyhalvorson committed:
add some gd hover styles already

Commit:
Display sketch ui kit slices

also provide proper build origin for them based on dev or prod

Commit:
Add hasSlice bool for whether slice.png exists

- and only for the base component the slice is for

jeffkamo committed:
Add container class

zacharyhalvorson committed:
hover effects

zacharyhalvorson committed:
spacing tweak

Commit:
Support for prod/dev build origins

henryh15 committed:
updating code props table css

henryh15 committed:
fixing className declaration

henryh15 committed:
css formatting

henryh15 committed:
css formatting

henryh15 committed:
css test

henryh15 committed:
css tweaks

henryh15 committed:
css tweaks

henryh15 committed:
css tweaks

henryh15 committed:
hiding second and third columns in prop table

henryh15 committed:
hiding 2nd/3rd column in table body

henryh15 committed:
typo on styles
